### PR TITLE
GG-39004 Node will dump persistence files on data corruption error

### PR DESCRIPTION
**Pull request by Erixonich**
https://ggsystems.atlassian.net/browse/GG-39004

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java [hunk 1]
```diff
@@ -1261,6 +1263,36 @@ public int pageSize() {
...
+
+            U.copy(new File(idxStore.getFileAbsolutePath()), new File(dumpDir, INDEX_FILE_NAME), false);
+
+            Set<Integer> parts = new HashSet<>();
+            for (long pageId : pageIds) {
+                parts.add(PageIdUtils.partId(pageId));
+            }
+
+            for (Integer part : parts) {
+                if (part >= pageStores.partStores.length)//skip index parts
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java:1284]: I don't understand this condition and this comment, can you elaborate?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java [hunk 2]
```diff
@@ -3236,6 +3236,57 @@ else if (file.exists()) {
...
+            boolean[] copied = new boolean[(int) (currIdx - lastCpIdx + 1)];
+
+            List<File> walFiles = new ArrayList<>();
+
+            if (archiver != null)
+                walFiles.addAll(Arrays.asList(walArchiveDir.listFiles(WAL_SEGMENT_COMPACTED_OR_RAW_FILE_FILTER)));
+
+            walFiles.addAll(Arrays.asList(walWorkDir.listFiles(WAL_SEGMENT_COMPACTED_OR_RAW_FILE_FILTER)));
+
+            if (archiver != null) {
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java:3261]: What if archiver is null?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java [hunk 3]
```diff
@@ -3236,6 +3236,57 @@ else if (file.exists()) {
...
+
+                            U.copy(file, new File(dumpDir, fileName(idx)), false);
+                        }
+                    }
+                }
+            }
+        } catch (Exception e) {
+            log.error("Failed to dump wal files", e);
+        }
+
```
Comment (ibessonov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java:3287]:
```suggestion
```

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java (hunk 2 above)
Comment (Erixonich) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/wal/FileWriteAheadLogManager.java:3261]: Then we don't need to check archived segments, don't we?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java (hunk 1 above)
Comment (Erixonich) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/file/FilePageStoreManager.java:1284]: Sure, if error occurred in reading index file, then part = 65535, which outbound of data partitions values.

**Review by ibessonov — APPROVED** [review]

**Pull request merged**